Release 0.3.0

miniche opened the pull request:
This release includes 4 new features, 2 improvements and 3 fixes.
Please note that this release break the compatibility with Python 2.5: only Python 2.6 or 2.7 are supported yet.

The release notes can be read on our official documentation: http://docs.centralreport.net/unix-agent/releases/0.3.0

**All commits made since the version 0.2.0:**
https://github.com/CentralReport/CentralReport/compare/master...release-0.3.0
## New features
- [x] #58: CentOS distribution support
- [x] #67: Travis CI and silent installer/uninstaller
- [x] #70: New online installer and uninstaller
- [x] #82: New CLI Manager
## Major improvements
- [x] #72: Added a check for the webserver port availability
- [x] #74: Bash functions cleaned
## Fixes
- [x] #73: Removed old config.py script
- [x] #79: Removed support for Python 2.5 and older
- [x] #81: Removed flot.js library
- [x] #84: Fixed "sudo -n" and "sudo -v" for OS X Mavericks
# Todo before merging:
- [x] Update the version number in the configuration
- [x] Add the release date in the NEWS file
- [x] Update the release notes (http://docs.centralreport.net/unix-agent/releases/0.3.0)
- [x] Generate the packages (installer/uninstaller)

Guimove committed:
Can now detect if it's a CentOS or Arch Linux

Guimove committed:
New detection of linux distrib

Guimove committed:
Added support of CentOS (and RedHat)

Guimove committed:
Cleanup and better detection

Guimove committed:
Update README.md with Travis status

Guimove committed:
Adding travis.yml to see how it works

Guimove committed:
Relauch travis build

Guimove committed:
Correction of the installation script path

Guimove committed:
install is only lauched by root user

Guimove committed:
autoinstall mode

Guimove committed:
autoinstall mode v2

Guimove committed:
Travis now build the install and uninstall scripts

Guimove committed:
travis.sh path correction

Guimove committed:
Travis script rights experiments

Guimove committed:
Correction of travis.sh not found

Guimove committed:
Correction of travis.sh not found

Guimove committed:
Add more verbosity to travis tests

miniche committed:
Updated "packager" with new bash functions

miniche committed:
Updated online installer/uninstaller

New URLs and little fixes

Guimove committed:
Update README.md

miniche committed:
Travis implementation improvements + Silent installer/uninstaller

miniche committed:
Fixed Travis script with the new "-s" argument for silent installer/uninstaller

miniche committed:
Spelling fixes

miniche committed:
Removed the .travis.sh script

The before_script and after_script options in the travis configuration are used instead.

Guimove committed:
Centos integration seems to be fully working

Guimove committed:
Check disk from a tuple

miniche committed:
Improved travis.yml with comments

miniche committed:
Merge pull request #71 from CentralReport/feature-travis-improvements

Travis implementation improvements

miniche committed:
Merge remote-tracking branch 'origin/develop' into feature-travis-in-readme

Conflicts:
	uninstall.sh

miniche committed:
Updated NEWS with the Travis integration and the silent installer/uninstaller

miniche committed:
Fixed spelling issues

Removed useless comments in the installer and in the uninstaller

miniche committed:
Fixed spelling

miniche committed:
Merge pull request #67 from CentralReport/feature-travis-in-readme

Travis CI integration and silent installer/uninstaller

miniche committed:
Merge remote-tracking branch 'origin/develop' into feature-online-installer

miniche committed:
Added a check for the webserver port availability

A function in the new "web" module checks if a port is used (on the local or a remote computer). In this case, the webserver is not loaded and an error is written in logs.

miniche committed:
Updated NEWS with the "port check" improvement

miniche committed:
Removed useless config.py script

This script is not used anymore with versions 0.2.0+, and can't work with new security features.
This will improved in the feature with a manager in CLI (but not now :P)

miniche committed:
Updated NEWS (removed config.py script)

miniche committed:
Added whitelists and blacklists for the packager (installer/uninstaller)

- Two packages are now generated: a package for the installer and a package for the uninstaller.
- The whitelist is read before the blacklist. In fact, all items in the blacklist are deleted in the files/folders copied by the whitelist.

The lists can contain:
- Files
- Folders (all files and sub-folders in the folder will be copied)

The blacklists can also contain items starting by *. A search in all directories will be performed and all corresponding elements will be deleted (e.g. for all *.pyc files)

miniche committed:
Moved "packager.sh" script to "make_packages.sh" in the packager folder

miniche committed:
Cleaned bash functions with a "return" statement

miniche committed:
Removed an old TODO in the bash functions (related to config.py script)

miniche committed:
Added docstring and refactored return values for all bash functions

All functions have now docstring (except launch_config_assistant, deleted in #73).
The return values have been refactored: each error returns is own error code instead of the sub-command error code.

miniche committed:
Refactored variable contents in the bash function "create_cr_user" (#69)

miniche committed:
Spelling fixes

miniche committed:
Updated bash docstrings

(spelling fixes)

miniche committed:
Updated NEWS for #74 (Bash function cleaned)

miniche committed:
Merge pull request #73 from CentralReport/fix-config-script

Removed useless config.py script

miniche committed:
Merge remote-tracking branch 'origin/develop' into feature-port-check

Conflicts:
	NEWS

miniche committed:
Merge pull request #72 from CentralReport/feature-port-check

Added a check for the webserver port availability

For the issue #65.

miniche committed:
Merge remote-tracking branch 'origin/develop' into clean-bash

Conflicts:
	NEWS
	bash/functions.inc.sh

miniche committed:
Merge pull request #74 from CentralReport/clean-bash

Cleaned bash code. Solves #68 and #69.

miniche committed:
Fixed header in whitelist/blacklist for the installer and the uninstaller

miniche committed:
Installer/Uninstaller: Added more checks and cleaned comments

miniche committed:
The packager have now its own functions library

It's cleaner and we can now detect any error during the process, and cleaning the "/tmp" folder at the end.

miniche committed:
Improved the online installer/uninstaller: comments and variables.

miniche committed:
Added online scripts in the package directory

It will be more easy to find them on copy them on a remote server after the package creation.

Guimove committed:
Merge branch 'develop' into feature-more-host-determination

# By Charles-Emmanuel (18) and others
# Via Charles-Emmanuel
* develop: (32 commits)
  Updated NEWS for #74 (Bash function cleaned)
  Updated bash docstrings
  Spelling fixes
  Refactored variable contents in the bash function "create_cr_user" (#69)
  Added docstring and refactored return values for all bash functions
  Removed an old TODO in the bash functions (related to config.py script)
  Cleaned bash functions with a "return" statement
  Updated NEWS with the "port check" improvement
  Updated NEWS (removed config.py script)
  Removed useless config.py script
  Added a check for the webserver port availability
  Fixed spelling
  Fixed spelling issues
  Updated NEWS with the Travis integration and the silent installer/uninstaller
  Improved travis.yml with comments
  Removed the .travis.sh script
  Fixed Travis script with the new "-s" argument for silent installer/uninstaller
  Travis implementation improvements + Silent installer/uninstaller
  Update README.md
  Add more verbosity to travis tests
  ...

Conflicts:
	bash/functions.inc.sh
	centralreport/centralreport.py
	uninstall.sh

Guimove committed:
Comment detection of OS other than supported

Guimove committed:
Correct mistake of my previous branch merge

Guimove committed:
Fix return code mistakes

Guimove committed:
Remove error when init file is not found (false positive error)

Guimove committed:
Better detection of disks

miniche committed:
Removed support for Python 2.5

A check has been added at the beginning of the installer and before starting the python daemon.

miniche committed:
Updated NEWS for the dropping of Python 2.5

Guimove committed:
Updating NEWS file and vars

Guimove committed:
Removed support of python < 2.5 in OS detection

Guimove committed:
Merge pull request #58 from CentralReport/feature-more-host-determination

CentOS distributions Support

miniche committed:
Fixed Python 2.5 detection

miniche committed:
Merge pull request #79 from CentralReport/remove-python-2.5

Removed support of Python 2.5

miniche committed:
Merge remote-tracking branch 'origin/develop' into feature-online-installer

miniche committed:
Added CentOS support in the online installer/uninstaller

miniche committed:
Updated NEWS file with the online installer/uninstaller feature

miniche committed:
Added a CLI Manager using the Urwid library

miniche committed:
Spelling fixes in the online installer part

miniche committed:
Updated URL for the online installer and uninstaller

miniche committed:
Removed Flot.js library (not used anymore) and the old "dashboard_mac" template.

miniche committed:
Updated NEWS for the PR #81 (removed flot.js library)

miniche committed:
Merge pull request #81 from CentralReport/remove-flot-js

Removed Flot.js library

miniche committed:
Added docstrings in the CLI Manager functions

miniche committed:
The Urwid library is now added to the sys.path in the main centralreport.py module

miniche committed:
Added a "DialogBox" object in the CLI module

Used to interact with the user, displaying texts or errors.

miniche committed:
Linked "Config" class to the CLI Manager

miniche committed:
Improved Dialog object in the "cli" module

miniche committed:
Improved CLI Manager with major improvements

- The manager must be run as root to perform some tasks (write new configuration, restart the daemon, …)
- CentralReport must be already installed on the current host to run the CLI Manager
- The configuration for the internal web server is now written in the configuration file
- The daemon restart when the user exit the CLI Manager to use the new configuration

PS: We must work on the "feature-webservices" branch to finish this PR for the "Online configuration" part.

miniche committed:
[WIP] Working on CLI Manager (MainCli, palette colors, Update the configuration, …)

miniche committed:
Improved CLI screens

miniche committed:
Major improvements in the CLI Manager

- The port for the internal web server is now checked when the user choose one new
- The main screen is now refreshed when the user modifies the configuration
- The daemon status ("CentralReport is not running" / "CentralReport is running with PID xxx") is now updated when the user starts/stops/restarts the daemon
- The daemon status is modified when the daemon is starting/stopping ("Starting CentralReport…" / "Stopping CentralReport…" / "Restarting CentralReport…")

Tech:
- The urwid main_loop is now located in the "WindowCli" class. It's more logical and can be updated directly within the object.

miniche committed:
Added "manage" option to the binary script for the CLI Manager

Now, we can start the CLI Manager with the command "centralreport manage" (or "/usr/local/bin/centralreport manage").

miniche committed:
The CLI wizard is now started at the end of the installation

miniche committed:
Removed all references to CentralReport Online

This part will be added in a future release.

miniche committed:
In the binary file, the "manager" parameter is now used to start… the CLI Manager.

(Was "manage" before)

miniche committed:
Updated NEWS file for #82 (CLI Manager)

miniche committed:
Minor fixes and improvements

miniche committed:
Merge pull request #70 from CentralReport/feature-online-installer

New online installer/uninstaller

miniche commented:
The pull request #70 (New online installer/uninstaller) has been merged.
Waiting for #82.

miniche committed:
Merge remote-tracking branch 'origin/develop' into feature-manager

Conflicts:
	NEWS

miniche committed:
Minor spelling fixes

miniche committed:
Merge pull request #82 from CentralReport/feature-manager

CLI Manager

miniche commented:
Hey guys!

The pull request #82 have been merged. So, we are ready to release the version 0.3.0!
I'm waiting for your ultimate review :)

The release notes have been updated on the documentation (http://docs.centralreport.net/unix-agent/releases/0.3.0).

I will update the last two points tonight:
- Update the version number in the configuration
- Add the release date in the NEWS file

Any comment is welcome :)

miniche committed:
Updated version number to 0.3.0.

Guimove commented:
I'll attempt to test this next week. 

I'm busy with all my interviews on my news job in Paris.

Ninir commented:
@Guimove Liar, you already signed your contract x)

Guimove commented:
Yes but I continu to met people like the client to knoledge.

miniche commented:
Hey ! Don't forget this :P

Ninir commented on bash/functions.inc.sh:
```diff
@@ -11,26 +11,45 @@
+    if [ "${CURRENT_OS}" == $"{OS_MAC}" ]; then
+        execute_privileged_command launchctl load -w ${STARTUP_PLIST}
+    else
+        execute_privileged_command /usr/local/bin/centralreport start
```
After looking at the code, I would store a reference to this binary path in order to avoid issues in the name or so. What do you think?

miniche replied:
Alright. And... it's already saved in the vars.inc.sh file. I will replace all occurrences with these variables.

Ninir commented on bash/functions.inc.sh:
```diff
@@ -425,12 +557,12 @@ function create_cr_user(){
             local fnumber=${fnumber_work_backwards_from}
             local user_id=0
-            until [ ${continue} = "yes" ] ; do
+            until [ ${continue} = true ] ; do
```
`until [ ${continue} ] ; do` ? :D

miniche replied:
Nop! In bash, "until" checks if the variable is...false. In fact, bash read it as "while this var is false", even if you write "until this var is false"...
(http://tldp.org/LDP/Bash-Beginners-Guide/html/sect_09_03.html)

Ninir commented on bash/functions.inc.sh:
```diff
@@ -441,11 +573,11 @@ function create_cr_user(){
                 fi
 
-                if [ ${number_used} = "true" ] ; then
+                if [ ${number_used} = true ] ; then
```
`if [ ${number_used} ] ; then`? :D

Ninir commented:
Seems good :) :+1:

miniche committed:
Minor improvements in bash scripts

miniche committed:
Fixed "sudo -n" and "sudo -v" for OS X Mavericks

On OS X 10.9 Mavericks, the command "sudo -n" return 0, even if the sudo session is not alive.
Replaced by "sudo -v -p", with a custom message.

miniche committed:
Updated "NEWS" file with #84 (Fixed "sudo -n" and "sudo -v" for OS X Mavericks)

miniche committed:
Merge pull request #84 from CentralReport/sudo-mavericks

Fixed "sudo -n" and "sudo -v" for OS X Mavericks

miniche commented:
I have fixed a little incompatibility with OS X Mavericks in #84, with the "sudo" command behavior.

miniche commented:
I have noticed a unwanted behavior with the memory compression on OS X 10.9 Mavericks. Read more here: #85.
We must fix it in the next release (0.3.x or 0.4.0).

So, I have tested CentralReport with the following OS:
- OS X 10.8
- OS X 10.9 (works well, except for the issue #85)
- Debian 6
- Debian 7
- Ubuntu 12.10
- CentOS 6.4

I think we are ready to release this version!

@Guimove, do you agree? We are waiting for your approval :)

Guimove commented:
Tested on CentOS et Debian Ok for me.

Well done bros sorry for being late.

miniche committed:
Updated NEWS file for the release 0.3.0

miniche commented:
The version 0.3.0 is ready! This branch will be merged in master in the next minutes.

Read more about this version on our blog: http://blog.centralreport.net
You can find the release notes on the documentation: http://docs.centralreport.net/unix-agent/releases/0.3.0

Happy monitoring!

miniche commented:
PS: The packages (installer and uninstaller) have been generated.
They are available at http://static.centralreport.net/ (please consult the documentation for more details: http://docs.centralreport.net/unix-agent/install)
